core/fault/stacktrace/pack.go

```diff
@@ -0,0 +1,374 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stacktrace
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/google/gapid/core/data/binary"
+)
+
+// Tweakables
+const (
+	bitChunkMinSize   = 2
```

Comment:
Notes:
 - Just increments might not be optimal.  If you build a histogram of the values, you can probably calculate the optimal chunk sizes.
 - Different optimal values might apply to the back-refences and to the deltas.
 - Are all of the addresses aligned by any chance?  (i.e. can we divide all by common factor?)
 - Finally, I do not think it is really worth the time to investigate anything I just said :-D

Reply:
> Just increments might not be optimal. If you build a histogram of the values, you can probably calculate the optimal chunk sizes.

I actually tried this - storing the table outweighed the costs for this case. I was surprised too.

> Are all of the addresses aligned by any chance? (i.e. can we divide all by common factor?)

Nope. Again, something that surprised me. My guess is that these aren't real PC addresses, but contain other metadata somehow.